drivers/led/ncp5623.c

```diff
@@ -149,7 +149,7 @@ static int ncp5623_led_init(const struct device *dev)
 			return -ENODEV;
 		}
 
-		if (led_info->num_colors != 3 || led_info->num_colors != 1) {
+		if (led_info->num_colors != 3 && led_info->num_colors != 1) {
```

Comment:
Thanks for spotting this.

In addition I think the code should rather be:

```
if (led_info->num_colors != NCP5623_CHANNEL_COUNT) {
                LOG_ERR("%s: invalid number of colors %d (must be %d when defining a single LED)",
                                  dev->name, led_info->num_colors, NCP5623_CHANNEL_COUNT);
}
```

@PetervdPerk-NXP, please can you have a look ?

Reply:
It was for the weird edge where one would be attaching one singe color led to the ncp5623.
@simonguinot suggestion to just enforce 3 colors in single led rgb mode would make more sense.

Reply:
Fixed proposed way.